### PR TITLE
Allow intercepting bot requests with custom criteria

### PR DESCRIPTION
**Pull request by jeremy-israel**

### CODE REVIEW

File: lib/datadome/configuration.rb [hunk 1]
```diff
@@ -9,13 +9,13 @@ def initialize
       @api_server = "api.datadome.co"
       @exclude_matchers = []
       @include_matchers = []
-      @monitor_mode = false
```
Comment (misdoro) [inline lib/datadome/configuration.rb:12]: Can we keep both maybe?

File: lib/datadome/configuration.rb [hunk 2]
```diff
@@ -9,13 +9,13 @@ def initialize
...
+      @intercept_matchers = []
```
Comment (misdoro) [inline lib/datadome/configuration.rb:12]: Do you mind adding some documentation on how to use it?

File: lib/datadome/configuration.rb (hunk 1 above)
Comment (jeremy-israel) [reply lib/datadome/configuration.rb:12]: With below logic for instance ?

Intercept if 
```ruby
(bot request) and (monitor mode disabled OR intercept match ?)
```

File: lib/datadome/configuration.rb (hunk 2 above)
Comment (jeremy-israel) [reply lib/datadome/configuration.rb:12]: good point

File: lib/datadome/configuration.rb (hunk 1 above)
Comment (misdoro) [reply lib/datadome/configuration.rb:12]: yes

Comment (jeremy-israel) [reply lib/datadome/configuration.rb:12]: done.

File: lib/datadome/configuration.rb (hunk 2 above)
Comment (jeremy-israel) [reply lib/datadome/configuration.rb:13]: done.

**Pull request merged**